src/Fhir.Anonymizer.Core/AnonymizerEngine.cs

```diff
@@ -33,25 +36,38 @@ public AnonymizerEngine(AnonymizerConfigurationManager configurationManager)
             _logger.LogDebug("AnonymizerEngine initialized successfully");
         }
 
-        public string AnonymizeJson(string json, bool isPrettyOutput = false)
+        public string AnonymizeJson(string json, bool isPrettyOutput = false, bool validateInput = false, bool validateOutput = false)
         {
             EnsureArg.IsNotNullOrEmpty(json, nameof(json));
 
             ElementNode root;
+            Resource resource;
             try
             {
-                root = ElementNode.FromElement(_parser.Parse(json).ToTypedElement());
+                resource = _parser.Parse<Resource>(json);
+                root = ElementNode.FromElement(resource.ToTypedElement());
             }
             catch(Exception innerException)
             {
                 throw new Exception("Failed to parse json resource, please check the json content.", innerException);
             }
 
+            if (validateInput)
+            {
+                _validator.ValidateInput(resource);
```

Comment:
I assume we do not fail the process here. If we fail the process here, users need to run the command again after fixing a single case. This can happen many times if there are many invalid files. If we do not fail the process here, users can get the validation report of the whole data set at once and fix them together.

Reply:
By offline discussion, we decide to fail the process fast. If the resource is invalid, ResourceNotValidException is thrown by engine. Outside tool catches the exception and continues processing. Exception message can be seen with verbose both enabled or disabled. Details are only given with verbose enabled.